pkg/tsdb/azuremonitor/azuremonitor-resource-handler.go

```diff
@@ -33,30 +33,31 @@ func (s *httpServiceProxy) writeErrorResponse(rw http.ResponseWriter, statusCode
 	rw.Header().Set("Content-Type", "application/json")
 	rw.WriteHeader(statusCode)
 
+	// Set error response to initial error message
+	errorBody := map[string]string{"error": message}
+
 	// Attempt to locate JSON portion in error message
 	re := regexp.MustCompile(`\{.*?\}`)
 	jsonPart := re.FindString(message)
-
-	var jsonData map[string]interface{}
-	if unmarshalErr := json.Unmarshal([]byte(jsonPart), &jsonData); unmarshalErr != nil {
-		errorMsg, _ := json.Marshal(map[string]string{"error": "Invalid JSON format in error message"})
-		_, err := rw.Write(errorMsg)
-		if err != nil {
-			return fmt.Errorf("unable to write HTTP response: %v", err)
+	if jsonPart != "" {
+		var jsonData map[string]interface{}
+		if unmarshalErr := json.Unmarshal([]byte(jsonPart), &jsonData); unmarshalErr != nil {
+			errorBody["error"] = fmt.Sprintf("Invalid JSON format in error message. Raw error: %s", message)
+			s.logger.Error("failed to unmarshal JSON error message", "error", unmarshalErr)
+		} else {
+			// Extract relevant fields for a formatted error message
+			errorType, _ := jsonData["error"].(string)
+			errorDescription, _ := jsonData["error_description"].(string)
+			if errorType == "" {
+				errorType = "UnknownError"
+			}
+
+			errorBody["error"] = fmt.Sprintf("%s: %s", errorType, errorDescription)
 		}
-		return unmarshalErr
-	}
-
-	// Extract relevant fields for a formatted error message
-	errorType, _ := jsonData["error"].(string)
-	errorDescription, _ := jsonData["error_description"].(string)
-	if errorType == "" {
-		errorType = "UnknownError"
 	}
-	formattedError := fmt.Sprintf("%s: %s", errorType, errorDescription)
 
-	errorMsg, _ := json.Marshal(map[string]string{"error": formattedError})
-	_, err := rw.Write(errorMsg)
+	jsonRes, _ := json.Marshal(errorBody)
```

Comment:
there will be no error body if `jsonPart` is empty. Is this the expected behavior?

Reply:
The error body will default to the initial error message passed into the function (see line 37)